File: src/main/java/com/hdup/DataGenerator.java
package com.hdup;

import java.util.Random;

/**
 * @author Vadim Bobrov
 */
public class DataGenerator {

    private static final String[] CUSTOMERS = new String[20];
    private static final String[] LOCATIONS = new String[] {"location0", "location1"};
    private static final String[] WIREIDS = new String[300];

    static {
        for (int i = 0; i < 20; i ++)
            CUSTOMERS[i] = "customer" + i;
        for (int i = 0; i < 300; i ++)
            WIREIDS[i] = "wire" + i;
    }

    private final Random random = new Random();

    public String getRandomCustomer() {
        return CUSTOMERS[random.nextInt(20)];
    }

    public String getCustomer(int i) {
        return CUSTOMERS[i];
    }

    public String getRandomLocation() {
        return LOCATIONS[random.nextInt(2)];
    }

    public String getLocation(int i) {
        return LOCATIONS[i];
    }

    public String getRandomWireId() {
        return WIREIDS[random.nextInt(300)];
    }

    public String getWireId(int i) {
        return WIREIDS[i];
    }

    public long getRandomMeasurement() {
        return random.nextLong();
    }


}

File: src/test/scala/com/outsmart/DataGenerator.scala
package com.outsmart

import scala.util.Random

/**
 * @author Vadim Bobrov
*/
class DataGenerator {

  private val CUSTOMERS =  new Array[String](20)
  private val LOCATIONS =  Array[String]("location0", "location1")
  private val WIREIDS =  new Array[String](300)

  for (i <- 0 until 20)
    CUSTOMERS(i) = "customer" + i

  for (i <- 0 until 300)
    WIREIDS(i) = "wireid" + i

  private val random : Random = new Random()

  def getRandomCustomer = CUSTOMERS(random.nextInt(20))
  def getRandomLocation = LOCATIONS(random.nextInt(2))
  def getRandomWireId = WIREIDS(random.nextInt(300))
  def getRandomMeasurement = random.nextLong()

  def getCustomer(i : Int) = CUSTOMERS(i)
  def getLocation(i : Int) = LOCATIONS(i)
  def getWireId(i : Int) = WIREIDS(i)

}
